test/e2e/tls_util_test.go

```diff
@@ -197,6 +200,32 @@ func TestNoneOfCaAndAppSecretExist(t *testing.T) {
 	verifyCASecret(t, caSecret, namespace)
 }
 
+// TestCustomCA ensures that if a user provides a custom Key and Cert and the CA and Application TLS assets
+// do not exist, the GenerateCert method can use the custom CA to generate the TLS assest.
+func TestCustomCA(t *testing.T) {
+	f := framework.Global
+	ctx := f.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cg := tlsutil.NewSDKCertGenerator(f.KubeClient)
+
+	customConfig := &tlsutil.CertConfig{
+		CertName: certName,
+		CAKey:    "testdata/ca.key",
+		CACert:   "testdata/ca.crt",
+	}
+	appSecret, _, _, err := cg.GenerateCert(newDummyCR(namespace), newAppSvc(namespace), customConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	verifyAppSecret(t, appSecret, namespace)
```

Comment:
We need to check that no ca is being put into k8s cluster.

Reply:
Do you mean that we should explicitly try to 'GET' the ca secret and Configmap with the given name and ensure nothing is returned?

Reply:
yes, that's correct.

Reply:
done